### DISCUSSION TITLE
Missing Linux-based authentication for code-server feature

### DISCUSSION THREAD

**Discussion by farshadf** [coder/code-server#5293] [General]
Hi,

I am trying to understand why there is no native linux-based authentication implemented on code-server. Works from other groups to add additional authentication opportunities to code-server make it very difficult to setup and use, and do not provide the benefits of using the standard user accounts in linux. Am I missing that this feature is somehow implemented already? If not, can somebody please advice on why such a key feature is missing? The community's help is much appreciated.

Best,
Farshad

**Comment by jsjoeio** [comment]
Can you please explain what you mean when you say "native linux-based authentication"?

What specifically are you looking for?

**Reply by farshadf** [reply]
Thank you for your prompt attention. I mean user and group administration available in most linux distros. In the case of RHEL and CentOS : [https://www.redhat.com/sysadmin/linux-user-group-management](https://www.redhat.com/sysadmin/linux-user-group-management).

The goal is to use the built-in user administration capability to control user access to the code-server.

**Reply by jsjoeio** [reply]
Hmm...I am not super familiar with that, but I imagine it would be the same as if you were using built-in user administration for a Python web server running on port 3000. 

As for built-in authentication for code-server, we have some information around passwords here: https://coder.com/docs/code-server/latest/FAQ#can-i-store-my-password-hashed

Let me know if that helps!

**Reply by farshadf** [reply]
Thanks for your response. Unfortunately, it does not seem to be using the built-in user administration like Python and it is not being integrated. For each user, we have to have a separate instance, or use other versions of code-server developed by third parties that provide multi-user option, which again none of them is integrated with the Linux's built-in user administration as far as I could find. 

If you have additional recommendations or if there is somebody else on the team with direct involvement with the authentication function, please let me know.

Best,
Farshad

**Reply by jsjoeio** [reply]
Have you taken a look at [coder/coder](https://github.com/coder/coder)? code-server was built with one user in mind, but coder was built for teams. 

Now it won't use the implementation you're looking for but should get you to the end goal - code-server for individual users and authentication for multiple users.
